### PR TITLE
AJAX Flash Header support (if AJAX HTTP request contains "X-Get-Flash: yes")

### PR DESCRIPTION
**Pull request by Ali1**
At times, it would be useful if flash messages could also be sent with AJAX responses. These would need to be handled client side with javascript notifying the user of the message.

### CODE REVIEW

**Comment by Ali1** [comment]
Example response header:
`X-Flash: {"flash":[{"message":"250.00 recorded with method: Cash","type":"success","params":[]}]}`

File: src/Controller/Component/FlashComponent.php [hunk 1]
```diff
@@ -40,11 +42,72 @@ class FlashComponent extends Component
...
+        $controller = $event->getSubject();
+
+        if (!$controller->getRequest()->is('ajax')) {
+            return null;
+        }
+
+        if (
+            !in_array('yes', array_map( // case insensitive
+                'strtolower',
+                $controller->getRequest()->getHeader('X-Get-Flash')
```
Comment (markstory) [inline src/Controller/Component/FlashComponent.php]: Could these extensions be done in a sub-class? Without other tools that support this header and its response header we end up with a bespoke implementation that nothing else supports.

Comment (Ali1) [reply src/Controller/Component/FlashComponent.php]: Yep - just wanted to canvas if there was interest in this in core.

**Comment by ADmad** [comment]
This additional complexity doesn't need to be in the core. It's a good candidate for a community plugin.

**Comment by dereuromark** [comment]
https://github.com/dereuromark/cakephp-flash/pull/8#issuecomment-586034562

**Pull request closed without merging**